arangod/RocksDBEngine/RocksDBEngine.cpp

```diff
@@ -795,8 +795,19 @@ void RocksDBEngine::start() {
   _options.compaction_readahead_size =
       static_cast<size_t>(opts._compactionReadaheadSize);
 
+  if (_createShaFiles) {
+    _checksumEnv.reset(NewChecksumEnv(rocksdb::Env::Default(), _path));
+    _options.env = _checksumEnv.get();
+    static_cast<checksum::ChecksumEnv*>(_checksumEnv.get())
+        ->getHelper()
+        ->checkMissingShaFiles();
+  } else {
+    _options.env = rocksdb::Env::Default();
+  }
+
```

Comment:
Is it safe to do it here? It looks like the encryption setup is done directly afterwards, in ` configureEnterpriseRocksDBOptions(...)`. 
I think we need to move the call to `checkMissingShaFiles()` directly after the `configureEnterpriseRocksDBOptions()` call.

Reply:
No, I think this is correct. Without encryption, we need this delegations:

  ChecksumEnv -> DefaultEnv     (if sha files should be generated)

and

  DefaultEnv    (if sha files should not be generated)

If encryption is on, we need

  EncryptingEnv -> ChecksumEnv -> DefaultEnv

That is, because we want the checksums of the **encrypted** data. Therefore, we must first set up the ChecksumEnv and **later** wrap the current env with the EncryptingEnv.